werft/build/job-config: Add withVM annotation if present on branch name

ArthurSens opened the pull request:
## Description
<!-- Describe your changes in detail -->
Since we can't pass annotations on commits anymore, this is needed to make Harvester load tests possible

## Related Issue(s)
<!-- List the issue(s) this PR solves -->
Fixes https://github.com/gitpod-io/ops/issues/1444

## How to test
<!-- Provide steps to test this PR -->
Create a new branch from this PR with `with-vm` present on its name 

Start a new werft job, it should trigger the creation of a Harvester VM.

Or just run `./dev/preview/test/load-test.sh`

## Release Notes
<!--
  Add entries for the CHANGELOG.md or "NONE" if there aren't any user facing changes.
  Each line becomes a separate entry.
  Format: [!<optional for breaking>] <description>
  Example: !basic auth is no longer supported
  See https://www.notion.so/gitpod/Release-Notes-513a74fdd23b4cb1b3b3aefb1d34a3e0
-->
```release-note
NONE
```

mads-hartmann commented on .werft/jobs/build/prepare.ts:
```diff
@@ -17,8 +18,23 @@ export async function prepare(werft: Werft) {
+}
+
+function lookForBranchNameWithVM(branchName: string): boolean {
+    return branchName.includes("with-vm") ||
```
Just just stick with `with-vm`. Currently the load-test is the only consumer of this so lets keep it simple ☺️

mads-hartmann commented on dev/preview/test/load-test.sh:
```diff
@@ -17,6 +17,7 @@ function log {
 START=1
 END=15
 BASE_BRANCH_NAME="vm-load-test"
+BRANCH_SUFFIX="withVM"
```
I'd prefer to stick with using hyphens and use `with-vm`

mads-hartmann replied:
especially now that the ts code looks for with-vm :)

mads-hartmann commented on dev/preview/test/load-test.sh:
```diff
@@ -52,19 +53,20 @@ y/n: " yn
+FULL_BRANCH_NAME="${BASE_BRANCH_NAME}-${BRANCH_SUFFIX}"
+log "Creating base branch ${FULL_BRANCH_NAME}"
+git checkout -b "${FULL_BRANCH_NAME}"
+git commit --allow-empty -m "Prepare load test"
```
We don't need to create a commit anymore now that we don't need to add the Werft commands ☺️

mads-hartmann commented on .werft/build.ts:
```diff
@@ -58,7 +58,7 @@ async function run(context: any) {
 
     await validateChanges(werft)
-    await prepare(werft)
+    config = await prepare(werft, config)
```
I would very much like to keep the config a const and not mutate the config options anywhere ☺️ It makes it easier to debug the the script if you keep the configuration parsing early and treat it as immutable for the rest of the job. So if possible, please move `lookForBranchNameWithVM` to job-config.ts and invoke the function there instead ☺️

mads-hartmann commented on .werft/jobs/build/prepare.ts:
```diff
@@ -17,8 +18,23 @@ export async function prepare(werft: Werft) {
         exec('gcloud container clusters get-credentials core-dev --zone europe-west1-b --project gitpod-core-dev');
-        werft.done('prep');
+
+        if (lookForBranchNameWithVM(config.repository.branch)) {
```
When you move this to job-config.ts please also log a line that says that we're overriding the value ☺️

mads-hartmann reviewed (changes requested):
Thanks! I have a few change requests before I can approve ☺️

ArthurSens commented:
@mads-hartmann I ended up changing the strategy completely.

I've just added another conditional when creating the job config

mads-hartmann reviewed (commented):
@ArthurSens The TS changes are much nicer, great job! I still have two unresolved comments though ☺️

ArthurSens commented:
Whoops, I guess I was a bit distracted when reading the review and missed a couple of things 😬 

I've addressed all comments, and also added a new flag for the load test script.

`--interval` can be useful here in case 30s is too fast or too slow for different scenarios

ArthurSens commented on dev/preview/test/load-test.sh:
```diff
@@ -5,7 +5,7 @@
 # Usage:
 #   ./dev/preview/test/load-test.sh
-#   ./dev/preview/test/load-test.sh --base-branch-name my-load-test-base-branch-name --start 1 --end 5
+#   ./dev/preview/test/load-test.sh --base-branch-name test --start 1 --end 5 --interval 30
```
I've changed the example branch name because of the 20 characters restriction 🤔

mads-hartmann reviewed (approved):
Awesome, LGTM. Added the hold label as CI hasn't completed running yet and don't want to mess with Tide :D

ArthurSens commented on dev/preview/test/load-test.sh:
```diff
@@ -16,7 +16,9 @@ function log {
+BRANCH_SUFFIX="with-vm"
 
 opts=$(getopt \
   --longoptions "start:,end:,base-branch-name:" \
```
```suggestion
  --longoptions "start:,end:,base-branch-name:,interval:" \
```

ArthurSens commented:
Sorry mads, just noticed something was missing 😅

mads-hartmann reviewed (approved):
🎉
